src/nav/navReducer.js

```diff
@@ -41,43 +34,10 @@ export const getStateForRoute = (route: string): NavigationState => {
   return state;
 };
 
-const rehydrate = (state, action) => {
-  // If there's no data to rehydrate, or no account data, show login screen.
-  if (!action.payload || !action.payload.accounts) {
-    return getStateForRoute('realm');
-  }
-
-  // If there are accounts but the active account is not logged in,
-  // show account screen.
-  const rehydratedState = action.payload;
-  if (!hasAuth(rehydratedState)) {
-    const { accounts } = rehydratedState;
-    return getStateForRoute(accounts && accounts.length > 1 ? 'account' : 'realm');
-  }
-
-  // If there's an active, logged-in account but no server data, then behave
-  // like `ACCOUNT_SWITCH`: show loading screen.  Crucially, `sessionReducer`
-  // will have set `needsInitialFetch`, too, so we really will be loading.
-  //
-  // (Valid server data must have a user: the self user, at a minimum.
-  // Compare getHaveServerData; this has an extra check because `users`
-  // may be missing entirely.)
-  if (rehydratedState.users === undefined || rehydratedState.users.length === 0) {
```

Comment:
The last part of this comment was meant to explain why this conditional can't simply use `getHaveServerData`. Since with this change we get to delete that bit of comment and the reason for it, we can also switch to simply using that selector :slightly_smiling_face: , like we do in `MainScreenWithTabs`. That'll make this code more explicit about what it's about.